util/cache/inmemory.go

```diff
@@ -29,6 +29,21 @@ func (i *InMemoryCache) Set(item *Item) error {
 	return nil
 }
 
+func (i *InMemoryCache) HasSame(key string, obj interface{}) bool {
+	var buf bytes.Buffer
+	err := gob.NewEncoder(&buf).Encode(obj)
+	if err != nil {
+		return false
+	}
+
+	bufIf, found := i.memCache.Get(key)
+	if !found {
+		return false
+	}
+	existingBuf := bufIf.(bytes.Buffer)
```

Comment:
Maybe we should safeguard this typecast, but I'm not sure. Can there be something else than `bytes.Buffer` in the in-memory cache for whatever reason?

Reply:
I think we want to panic here, because if In-memory cache has garbage we should restart pod. Added explicit check and panic with the meaningful message